### ISSUE TITLE
Feature request: Support node 20 - upgrade msal-node module

### ISSUE THREAD

**Issue by omeralon1** [OfficeDev/microsoft-365-agents-toolkit#10849]
**Request Details**
Trying to install teamsfx package on node 20 project with yarn, but the installation can't be done due to old deprecated version of msal-node package requirement (1.18.4)
This version supports node version 18 and below.
The error I get when trying to install:
`@azure/msal-node@1.18.4: The engine "node" is incompatible with this module. Expected version "10 || 12 || 14 || 16 || 18". Got "20.9.0"`
After this message yarn fails the installation.

**Solution Description**
Upgrade msal-node module to version 2+, which supports node 20.

**Alternatives Considered**
* Downgrade the whole project node version to 18
* Install using --ignore-engines flag
* Install using npm instead of yarn, which only warns but doesn't fail the installation

Will appreciate your help!

**Comment by eitanlevinzon-astrix** [comment]
+1

**Comment by MuyangAmigo** [comment]
Hey @omeralon1, @eitanlevinzon-astrix we are working on an upgrade of `teamsfx` to use the latest `msal-node` packages, please stay tunned.

**Comment by yiqing-zhao** [comment]
Hi @omeralon1, @eitanlevinzon-astrix, all Teams bots and message extensions are developed using the botbuilder-js SDK. Presently, we're eagerly anticipating support for Node.js 20 in the Botbuilder SDK. You can track the progress of this issue at [microsoft/botbuilder-js#4550](https://github.com/microsoft/botbuilder-js/issues/4550).

We've conducted tests to ensure that our current templates remain functional after upgrading to Node.js 20. As a temporary workaround for any warnings encountered, you can use "npm install --ignore-engines". We'll keep you informed and update here once we receive any feedback from the botbuilder team. Thanks!

**Comment by omeralon1** [comment]
Hi @yiqing-zhao 
Thank you for the answer
Will keep tracking 👀

**Comment by Kamahl19** [comment]
@yiqing-zhao @MuyangAmigo The botbuilder has updated msal to `@azure/msal-node` in their latest releast however the `@microsoft/teamsfx` still references the `"@azure/msal-node": "^1.14.6",` which is causing installation warning.

**Comment by yiqing-zhao** [comment]
> @yiqing-zhao @MuyangAmigo The botbuilder has updated msal to `@azure/msal-node` in their latest releast however the `@microsoft/teamsfx` still references the `"@azure/msal-node": "^1.14.6",` which is causing installation warning.

Hi @Kamahl19, we're currently working on upgrading TeamsFx SDK, and will update here once it’s merged and released. Thanks!

**Comment by JustGoscha** [comment]
Is it supposed to work though with `msal-node@2.x.x`?

**Comment by MuyangAmigo** [comment]
> Is it supposed to work though with `msal-node@2.x.x`?

Yes and a new stable version of `teamsfx` will be released this week.
